src/Schema/Validator.php

```diff
@@ -4,173 +4,15 @@
-                (new MaxProperties($this->schema))->validate($this->data, $this->schema->maxProperties);
-            }
-
-            if (isset($this->schema->minProperties)) {
-                (new MinProperties($this->schema))->validate($this->data, $this->schema->minProperties);
-            }
-
-            if (isset($this->schema->required)) {
-                (new Required($this->schema, $this->validationStrategy))->validate($this->data, $this->schema->required);
-            }
-
-            if (isset($this->schema->enum)) {
-                (new Enum($this->schema))->validate($this->data, $this->schema->enum);
-            }
-
-            // The following properties are taken from the JSON Schema definition but their definitions were adjusted to the OpenAPI Specification.
-            if (isset($this->schema->type)) {
-                (new Type($this->schema))->validate($this->data, $this->schema->type, $this->schema->format);
-            }
-
-            if (isset($this->schema->items)) {
-                (new Items($this->schema, $this->validationStrategy, $this->prepareBreadCrumb()))->validate($this->data, $this->schema->items);
-            }
-
-            if (isset($this->schema->properties) && count($this->schema->properties)) {
-                $additionalProperties = $this->schema->additionalProperties ?? null;
-                (new Properties($this->schema, $this->validationStrategy, $this->prepareBreadCrumb()))->validate(
-                    $this->data,
-                    $this->schema->properties,
-                    $additionalProperties
-                );
-            }
-
-            if (isset($this->schema->allOf) && count($this->schema->allOf)) {
-                (new AllOf($this->schema, $this->validationStrategy, $this->prepareBreadCrumb()))->validate($this->data, $this->schema->allOf);
-            }
-
-            if (isset($this->schema->oneOf) && count($this->schema->oneOf)) {
-                (new OneOf($this->schema, $this->validationStrategy, $this->prepareBreadCrumb()))->validate($this->data, $this->schema->oneOf);
-            }
-
-            if (isset($this->schema->anyOf) && count($this->schema->anyOf)) {
-                (new AnyOf($this->schema, $this->validationStrategy, $this->prepareBreadCrumb()))->validate($this->data, $this->schema->anyOf);
-            }
-
-            if (isset($this->schema->not)) {
-                (new Not($this->schema, $this->validationStrategy, $this->prepareBreadCrumb()))->validate($this->data, $this->schema->not);
-            }
-
-            //   ✓  ok, all checks are done
-        } catch (ValidationKeywordFailed $e) {
-            $e->hydrateDataBreadCrumb($this->prepareBreadCrumb());
-            throw $e;
-        }
-    }
-
-    protected function prepareBreadCrumb() : BreadCrumb
-    {
-        return $this->dataBreadCrumb ?? new BreadCrumb();
-    }
+    public function validate($data, Schema $schema, ?BreadCrumb $breadCrumb = null) : void;
```

Comment:
As a future idea i suggest to replace third argument to some kind of `ValidationContext` instead (not in this PR). WDYT?

Reply:
In general, yes, I support this idea. 
That would be a good thing when(if) there would be more arguments to pass in. 
If there is just one thing at the moment (a bread crumb), that looks ok-ish to me.

Reply:
Yea, thats why I call it `future` idea ) Not now for sure